src/core/thread/neighbor_table.cpp

```diff
@@ -271,7 +271,10 @@ void NeighborTable::Signal(Event aEvent, const Neighbor &aNeighbor)
         case kChildRemoved:
         case kChildModeChanged:
 #if OPENTHREAD_FTD
-            static_cast<Child::Info &>(info.mInfo.mChild).SetFrom(static_cast<const Child &>(aNeighbor));
+            if (Get<ChildTable>().Contains(aNeighbor))
```

Comment:
Is it valid to have a neighbor not in child table? Should we assert?

Reply:
I think we can assert here instead. 
All the calls to `NeighborTable::Signal()` with `kChild{Added/Removed/Updated}` are passing in `Child` entry.

Reply:
Added assert in new push